engine/immediate_task_conrrent_processor.go

```diff
@@ -277,13 +277,17 @@ func (w *immediateTaskConcurrentProcessor) applyStateFailureRecoveryPolicy(ctx c
 		}
 		nextImmediateTask := persistence.ImmediateTask{
 			ShardId:            task.ShardId,
-			TaskType:           persistence.ImmediateTaskTypeWaitUntil,
 			ProcessExecutionId: task.ProcessExecutionId,
 			StateExecutionId: persistence.StateExecutionId{
 				StateId:         *prep.Info.StateConfig.StateFailureRecoveryOptions.StateFailureProceedStateId,
 				StateIdSequence: 1,
 			},
 		}
+		if prep.Info.StateConfig.StateFailureRecoveryOptions.StateFailureProceedStateConfig.GetSkipWaitUntil() {
```

Comment:
I think technically `StateFailureProceedStateConfig` can be nil which could have NPE? 
maybe better to check the nil here:
```golang
proceedStateConfig := &AsyncStateConfig
if prep.Info.StateConfig.StateFailureRecoveryOptions.StateFailureProceedStateConfig != nil{
    ...
}
...
```